arangod/Aql/QuerySnippet.cpp

```diff
@@ -183,6 +184,21 @@ void CloneWorker::setUsedShardsOnClone(ExecutionNode* node,
         TRI_ASSERT(false);
         THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL_AQL);
       }
+    } else if (clone->getType() == ExecutionNode::JOIN) {
+      // JoinNodes handles multiple collections
+      auto joinNode = dynamic_cast<JoinNode*>(clone);
+      if (joinNode != nullptr) {
+        // we've found a Disjoint SmartGraph node, now add the `i` th shard
+        // for used collections
```

Comment:
This comment looks weird. 
```suggestion
        // found a JoinNode, now add the `i` th shard for used collections
```